Added Metal Replay ProcessChunk()

Zorro666 opened the pull request:
## Description

Added:
`bool WrappedMTLDevice::ProcessChunk(ReadSerialiser &ser, MetalChunk chunk);`

`SystemChunk::InitialContentsList` handling will be in a future PR.
`SystemChunk::CaptureEnd` handling is incomplete and will be completed in a future PR.

Added declarations to `WrappedMTLDevice` for dummy objects used for serialization replay.
```
WrappedMTLBuffer *m_DummyBuffer = NULL;
WrappedMTLCommandBuffer *m_DummyReplayCommandBuffer = NULL;
WrappedMTLCommandQueue *m_DummyReplayCommandQueue = NULL;
WrappedMTLLibrary *m_DummyReplayLibrary = NULL;
WrappedMTLRenderCommandEncoder *m_DummyReplayRenderCommandEncoder = NULL;
WrappedMTLBlitCommandEncoder *m_DummyReplayBlitCommandEncoder = NULL;
```
The dummy objects will be created in a future PR.

Zorro666 committed:
Added Metal Replay ProcessChunk()

bool WrappedMTLDevice::ProcessChunk(ReadSerialiser &ser, MetalChunk chunk)

SystemChunk::InitialContentsList is handled but not implemented

Added declarations for dummy objects used for serialization replay

WrappedMTLBuffer *m_DummyBuffer;
WrappedMTLCommandBuffer *m_DummyReplayCommandBuffer;
WrappedMTLCommandQueue *m_DummyReplayCommandQueue;
WrappedMTLLibrary *m_DummyReplayLibrary;
WrappedMTLRenderCommandEncoder *m_DummyReplayRenderCommandEncoder;
WrappedMTLBlitCommandEncoder *m_DummyReplayBlitCommandEncoder;

The dummy objects are not created in this commit.

baldurk commented on renderdoc/driver/metal/metal_core.cpp:
```diff
@@ -69,6 +72,365 @@ void WrappedMTLDevice::AddEvent()
+  {                                               \
+    RDCERR("MetalChunk::" #CHUNK " not handled"); \
+    return false;                                 \
+  }
```
This would be cleaner I think to have this macro only contain the body of the case or only the error message, and use the `chunk` parameter. Then the cases would look like:

```
      case MetalChunk::MTLDevice_newSharedTextureWithDescriptor:
        METAL_CHUNK_NOT_HANDLED();
      case MetalChunk::MTLDevice_newSharedTextureWithHandle:
        METAL_CHUNK_NOT_HANDLED();
      case MetalChunk::MTLDevice_newSamplerStateWithDescriptor:
        METAL_CHUNK_NOT_HANDLED();
```

That way the case statements are more uniform and you still don't repeat the enum value.

Zorro666 committed:
Add case for all MetalChunk in ProcessChunk switch

Add the case manually and remove it from the METAL_CHUNK_NOT_HANDLED() helper macro
PR Feedback
